src/components/user/PersonalProfileEditForm.js

```diff
@@ -29,6 +30,16 @@ export default function PersonalProfileEditForm() {
   const { user } = useAuth();
   const userData = user.user; //he user that's returned is nested in its original response shape so to use the actual user you'll need to unwrap it
 
+  useEffect(() => {
+    setFormData({
+      name: user?.username || '',
+      email: user?.email || '',
```

Comment:
Above on line 31, we store the user info in the `userData` variable.

But here you're trying to get `user?.username` which will turn out to be undefined. You can see this if you add a console log inside the useEffect, right between line 33 and line 34:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/12c0048d-b6e5-423b-b9a6-452b5b6d621a)

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/d3e6f241-3da0-43ee-a0a2-3a0c5ab94db6)

That is why you're seeing those errors because when we click "Save Changes" button, we try to submit the current value of the `formData`  variable. But we first check and set errors on line 50-54 if the `formData.name` or `formData.email` fields are empty.  Currently they are empty because we're not setting the values correctly at first.

If you change these lines to:

```suggestion
      name: userData?.username || '',
      email: userData?.email || '',
```

that will avoid that particular issue

Reply:
Thank you very much❤️ Have I understood correctly that  we obtain userData from useAuth hook here?